job-service-db/src/main/resources/procedures/getJobs.sql

```diff
@@ -173,6 +174,17 @@ BEGIN
          ELSE quote_ident(in_sort_field)
        END ||
         ' ' || CASE WHEN in_sort_ascending THEN 'ASC' ELSE 'DESC' END;
+
+    -- Create temporary table as a base to update the job progress
+    EXECUTE 'CREATE TEMPORARY TABLE getJobTemp ON COMMIT DROP AS ' || sql ;
+
+    FOR jobId IN SELECT * FROM getJobTemp LOOP
+            -- Process outstanding job updates
+            PERFORM internal_update_job_progress(in_partition_id, jobId);
+    END LOOP;
+
+
+    -- Execute and return the table created
     RETURN QUERY EXECUTE sql;
```

Comment:
You're now executing the `sql` twice.  What if things have changed by the second execution and it gives different jobs to the ones you just updated?  Shouldn't you just be returning the contents of your temporary table, with statuses updated of course?

Reply:
Statuses returned have not been updated.